litestar/_openapi/schema_generation/schema.py

```diff
@@ -274,14 +277,24 @@ def for_field_definition(self, field_definition: FieldDefinition) -> Schema | Re
 
         result: Schema | Reference
 
-        # NOTE: The check for whether the field_definition.annotation is a Pagination type
-        # has to come before the `is_dataclass_check` since the Pagination classes are dataclasses,
-        # but we want to handle them differently from how dataclasses are normally handled.
-        if field_definition.is_optional:
+        if _do_enum_schema(field_definition):
+            annotation = _type_or_first_not_none_inner_type(field_definition)
```

Comment:
I didn't use `make_non_optional_union()` like we do for literal types, b/c the literal schema generation has handling for flattening out nested literals, and the enum handling does not.

So a union like `EnumA | EnumB | None` is still going to be rendered as Object.